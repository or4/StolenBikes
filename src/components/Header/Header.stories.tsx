import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { Header } from './Header';

storiesOf('Header', module).add('Header', () => <Header />);
