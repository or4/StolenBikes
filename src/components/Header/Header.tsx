import React from 'react';
import styled from 'styled-components';

import logo from 'components/assets/logo.png';
import { HeaderTextContainer } from 'components/HeaderTextContainer';
import { media } from 'ui/media';

const Container = styled.div`
    display: flex;
    align-items: center;

	${media.mobile`
    	height: 60px;
  	`}

	${media.tablet`
    	height: 100px;
	`}

    ${media.desktop`
    	height: 140px;
	`}
`;

const Logo = styled.img`
    width: 64px;
    height: 64px;

    ${media.mobile`
		padding-right: 16px;
  	`}

    ${media.tablet`
		padding-right: 32px;
	`}

    ${media.desktop`
		padding-right: 32px;
	`}
`;

export const Header = () => (
    <Container>
        <Logo src={logo} />
        <HeaderTextContainer />
    </Container>
);
