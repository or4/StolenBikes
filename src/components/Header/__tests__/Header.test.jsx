import React from 'react';
import { mount } from 'enzyme';
import { Header } from '../Header';

describe('Header', () => {
    it('should render correctly', () => {
        const button = <Header />;

        const mounted = mount(button);
        expect(mounted).toMatchSnapshot();
        mounted.unmount();
    });
});
