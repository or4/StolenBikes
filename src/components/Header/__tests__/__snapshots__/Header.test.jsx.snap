// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<Header>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": true,
            "lastClassName": "jbeFzL",
            "rules": Array [
              "
    display: flex;
    align-items: center;

	",
              "
        @media ",
              "(max-width: ",
              "479",
              "px)",
              " {
            ",
              "
    	height: 60px;
  	",
              "
        }
    ",
              "

	",
              "
        @media ",
              "(min-width: ",
              "480",
              "px) and (max-width: ",
              "991",
              "px)",
              " {
            ",
              "
    	height: 100px;
	",
              "
        }
    ",
              "

    ",
              "
        @media ",
              "(min-width: ",
              "992",
              "px)",
              " {
            ",
              "
    	height: 140px;
	",
              "
        }
    ",
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bxivhb jbeFzL"
      >
        <styled.img
          src="logo.png"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": true,
                  "lastClassName": "bWdRmF",
                  "rules": Array [
                    "
    width: 64px;
    height: 64px;

    ",
                    "
        @media ",
                    "(max-width: ",
                    "479",
                    "px)",
                    " {
            ",
                    "
		padding-right: 16px;
  	",
                    "
        }
    ",
                    "

    ",
                    "
        @media ",
                    "(min-width: ",
                    "480",
                    "px) and (max-width: ",
                    "991",
                    "px)",
                    " {
            ",
                    "
		padding-right: 32px;
	",
                    "
        }
    ",
                    "

    ",
                    "
        @media ",
                    "(min-width: ",
                    "992",
                    "px)",
                    " {
            ",
                    "
		padding-right: 32px;
	",
                    "
        }
    ",
                    "
",
                  ],
                },
                "displayName": "styled.img",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": "img",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            src="logo.png"
          >
            <img
              className="sc-ifAKCX bWdRmF"
              src="logo.png"
            />
          </StyledComponent>
        </styled.img>
        <HeaderTextContainer>
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "bDWFJH",
                    "rules": Array [],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="sc-bdVaJa bDWFJH"
              >
                <styled.div>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bwzfXH",
                          "isStatic": true,
                          "lastClassName": "keAcTh",
                          "rules": Array [
                            "
    font-weight: 600;

	",
                            "
        @media ",
                            "(max-width: ",
                            "479",
                            "px)",
                            " {
            ",
                            "
    	font-size: 18px;
  	",
                            "
        }
    ",
                            "

	",
                            "
        @media ",
                            "(min-width: ",
                            "480",
                            "px) and (max-width: ",
                            "991",
                            "px)",
                            " {
            ",
                            "
    	font-size: 32px;
	",
                            "
        }
    ",
                            "

    ",
                            "
        @media ",
                            "(min-width: ",
                            "992",
                            "px)",
                            " {
            ",
                            "
    	font-size: 48px;
	",
                            "
        }
    ",
                            "
",
                          ],
                        },
                        "displayName": "styled.div",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bwzfXH",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <div
                      className="sc-bwzfXH keAcTh"
                    >
                      Police Department of Berlin
                    </div>
                  </StyledComponent>
                </styled.div>
                <styled.div>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": true,
                          "lastClassName": "cxvLWi",
                          "rules": Array [
                            "
	",
                            "
        @media ",
                            "(max-width: ",
                            "479",
                            "px)",
                            " {
            ",
                            "
    	font-size: 14px;
  	",
                            "
        }
    ",
                            "

	",
                            "
        @media ",
                            "(min-width: ",
                            "480",
                            "px) and (max-width: ",
                            "991",
                            "px)",
                            " {
            ",
                            "
    	font-size: 16px;
	",
                            "
        }
    ",
                            "

    ",
                            "
        @media ",
                            "(min-width: ",
                            "992",
                            "px)",
                            " {
            ",
                            "
    	font-size: 32px;
	",
                            "
        }
    ",
                            "
",
                          ],
                        },
                        "displayName": "styled.div",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <div
                      className="sc-htpNat cxvLWi"
                    >
                      Stolen bykes
                    </div>
                  </StyledComponent>
                </styled.div>
              </div>
            </StyledComponent>
          </styled.div>
        </HeaderTextContainer>
      </div>
    </StyledComponent>
  </styled.div>
</Header>
`;
